### PR TITLE
Additional stylus import features

### PR DESCRIPTION
**Pull request by m4dz**
This PR introduces extended @import features (see #10):
- module import : `@import 'foo'` > `foo/index.styl` if it exists
- globbing import : `@import 'glob/*'` > `[glob/file1.styl, glob/file2.styl,...]`

### CODE REVIEW

File: src/index.coffee [hunk 1]
```diff
@@ -86,7 +87,15 @@ progenyConstructor = (mode, settings = {}) ->
 		deps = []
 		dirs.forEach (dir) ->
 			paths.forEach (path) ->
-				deps.push sysPath.join dir, path
+				path = sysPath.join dir, path
+				basePath = path.slice 0, -(sysPath.extname(path).length)
+				if glob.hasMagic path
+					deps.push.apply deps, glob.sync path
```
Comment (es128) [inline src/index.coffee]: All of these sync methods are going to kill performance. Need to refactor to use only async methods.

**Comment by m4dz** [comment]
Now fully async :smile:.

File: src/index.coffee [hunk 2]
```diff
@@ -117,15 +118,33 @@ progenyConstructor = (mode, settings = {}) ->
...
+							return callback() if err
+							each files, (path, callback) ->
+								depsList.push path if potentialDeps
+								addDep path, depsList, callback
+							, callback
+					else
+						depsList.push path if potentialDeps
+						ext = ".#{extension}"
+						modulePath = path.replace ext, "#{sysPath.sep}index#{ext}"
+						fs.exists modulePath, (exists) ->
```
Comment (es128) [inline src/index.coffee]: The `fs.exists` call is redundant. Just add any path you want to check to the list, and if it doesn't exist when `readFile` happens it'll get pruned out.

File: src/index.coffee [hunk 3]
```diff
@@ -117,15 +118,33 @@ progenyConstructor = (mode, settings = {}) ->
...
+						glob path, (err, files) ->
+							return callback() if err
+							each files, (path, callback) ->
+								depsList.push path if potentialDeps
+								addDep path, depsList, callback
+							, callback
+					else
+						depsList.push path if potentialDeps
+						ext = ".#{extension}"
+						modulePath = path.replace ext, "#{sysPath.sep}index#{ext}"
```
Comment (es128) [inline src/index.coffee]: Checking for the index file should be exposed by a new option and added to the default settings for stylus.

File: src/index.coffee [hunk 4]
```diff
@@ -117,15 +118,33 @@ progenyConstructor = (mode, settings = {}) ->
 				if path in depsList
 					callback()
 				else
-					depsList.push path if potentialDeps
-					fs[mode].readFile path, encoding: 'utf8', (err, source) ->
-						return callback() if err
-						depsList.push path unless potentialDeps or path in depsList
-						parseDeps path, source, depsList, callback
+					if glob.hasMagic path
```
Comment (es128) [inline src/index.coffee]: Glob checking should also be behind a new option and enabled for specific formats that support it

**Comment by m4dz** [comment]
I made module and glob import now optional. Let me know if it's good for you :grinning:.

**Comment by es128** [comment]
Thanks, this is great!

There's a few more things I'll probably want to tweak about this implementation, but I may just merge and fiddle with it myself from there rather than continuing to go back and forth on the PR. Will get back to you.

Also, I'd be happy to add you as a maintainer to this project if you're interested in continuing to contribute to it and help with support.

**Comment by m4dz** [comment]
Thanks!

> I may just merge and fiddle with it myself from there

No problem. Tell me if some things aren't clear enough.

> I'd be happy to add you as a maintainer to this project

Sure! We intensively use Brunch and its plugins ecosystem at @cozy, so contribute and help to support its toolchain make sense for me. I'll be proud to be added as a maintainer to this project.

**Pull request merged**